web/src/lib/components/admin-page/settings/auth/auth-settings.svelte

```diff
@@ -126,16 +127,30 @@
 
               <SettingInputField
                 inputType={SettingInputFieldType.TEXT}
-                label={$t('admin.oauth_client_secret').toUpperCase()}
+                label="CLIENT_SECRET"
+                description={$t('admin.oauth_client_secret_description')}
                 bind:value={config.oauth.clientSecret}
-                required={true}
                 disabled={disabled || !config.oauth.enabled}
                 isEdited={!(config.oauth.clientSecret == savedConfig.oauth.clientSecret)}
               />
 
+              {#if config.oauth.clientSecret}
+                <SettingSelect
+                  label="TOKEN_ENDPOINT_AUTH_METHOD"
+                  bind:value={config.oauth.tokenEndpointAuthMethod}
+                  disabled={disabled || !config.oauth.enabled || !config.oauth.clientSecret}
```

Comment:
It's a bit weird that this is only shown when `clientSecret` is set, but you still disable it when it's not set. I'd only do either of them, out of which I'd prefer just disabling that field and not removing it.

Reply:
Id prefer hiding it if it doesn't apply. I'll just keep disabled in case it ever changes in the future.